### PR TITLE
Autocomplete input client

### PR DESCRIPTION
**Pull request by MaskaSaw**

### CODE REVIEW

File: hotels-backend/Hotels/Controllers/HotelsController.cs [hunk 1]
```diff
@@ -101,6 +101,24 @@ public async Task<ActionResult<IEnumerable<Room>>> GetRooms(int id)
                 .ToListAsync();
         }
 
+        //GET: api/Hotels/Countries
+        [HttpGet("Countries")]
+        public async Task<ActionResult<IEnumerable<string>>> GetCountries()
+        {
+            return await _context.Hotels
+                .Select(hotel => hotel.Country)
+                .ToListAsync();
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/HotelsController.cs]: What if there are multiple hotels in one country? Shouldn't you get only unique values here?

File: hotels-backend/Hotels/Controllers/HotelsController.cs [hunk 2]
```diff
@@ -101,6 +101,24 @@ public async Task<ActionResult<IEnumerable<Room>>> GetRooms(int id)
...
+        }
+
+        //GET: api/Hotels/Cities
+        [HttpGet("Cities")]
+        public async Task<ActionResult<IEnumerable<string>>> GetCities()
+        {
+            return await _context.Hotels
+                .Select(hotel => hotel.City)
+                .ToListAsync();
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/HotelsController.cs]: What if there are thousands of cities in your system?

File: hotels-client-frontend/src/app/hotels/hotels.component.ts [hunk 3]
```diff
@@ -47,9 +62,18 @@ export class HotelsComponent implements OnInit {
     return true;
   }
 
+  shuffle(array: Hotel[]): Hotel[] {
+    return array.sort(() => Math.random() -0.5);
+  }
```
Comment (andrei-kruhliakou-itechart) [inline hotels-client-frontend/src/app/hotels/hotels.component.ts]: Where do you use that?

Comment (MaskaSaw) [reply hotels-client-frontend/src/app/hotels/hotels.component.ts:67]: After getting cities and countries from the server, i sort them and then remove all the duplicates

File: hotels-backend/Hotels/Controllers/HotelsController.cs [hunk 4]
```diff
@@ -107,15 +107,18 @@ public async Task<ActionResult<IEnumerable<string>>> GetCountries()
...
             return await _context.Hotels
                 .Select(hotel => hotel.Country)
+                .Distinct()
                 .ToListAsync();
         }
 
         //GET: api/Hotels/Cities
         [HttpGet("Cities")]
-        public async Task<ActionResult<IEnumerable<string>>> GetCities()
+        public async Task<ActionResult<IEnumerable<string>>> GetCities([FromQuery] string country)
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/HotelsController.cs]: You can still get thousands of cities in large contries (e.g. USA)

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**